src/main.rs

```diff
@@ -61,15 +60,20 @@ fn main() -> anyhow::Result<()> {
 
     match args {
         SemverChecks::CheckRelease(args) => {
-            let loader = Box::new(baseline::RustdocBaseline::new(
-                args.baseline_rustdoc_path.clone(),
-            ));
+            let loader: Box<dyn baseline::BaselineLoader> =
+                if let Some(path) = args.baseline_rustdoc_path.as_deref() {
+                    Box::new(baseline::RustdocBaseline::new(path.to_owned()))
+                } else if let Some(root) = args.baseline_root.as_deref() {
+                    Box::new(baseline::PathBaseline::new(root.to_owned())?)
+                } else {
+                    unreachable!("a member of the `baseline` group must be present");
+                };
             let rustdoc = dump::RustDoc::new().deps(false).silence(false);
 
             let rustdoc_paths =
                 if let Some(current_rustdoc_path) = args.current_rustdoc_path.as_deref() {
                     vec![(
-                        loader.load_rustdoc("<unknown>")?,
+                        loader.load_rustdoc(&rustdoc, "<unknown>")?,
                         current_rustdoc_path.to_owned(),
                     )]
```

Comment:
If I'm reading this correctly, it seems like this is always generating new rustdoc JSON since new line 71 is always hit, even if there's a `current_rustdoc_path` set. That doesn't seem right to me?

Reply:
line 71 just creates the rustdoc wrapper.  The active `loader` is what determines whether rustdoc is generated or not.

Reply:
Changing `RustDoc` to `RustDocCommand` helps a lot here, thanks! To simplify the thought process of the next person to come across this, consider leaving a comment next to `let rustdoc` mentioning that the command is prepared but not executed, and that the loader gets to choose whether to run it or not.

Reply:
I renamed the variable to clarify this is the command.  This should clarify that it still needs a call done on it to actually execute.